### DISCUSSION TITLE
Input G/Gmax without HH Calibration

### DISCUSSION THREAD

**Discussion by xzzgameboy** [Caltech-geoquake/PySeismoSoil#18] [Q&A] [answered]
Hi,

We have tested G/Gmax of soils on a site from resonant column (small to medium) and hollow cylinder torsional shear testing (large strain). Is there a way that we can directly input the measured G/Gmax without involving HH-based strength correction since we believe the measured G/Gmax is the correct curve.

Any help is greatly appreciated.

Steve

**Comment by jsh9** [comment, marked as answer]
Hi @xzzgameboy , thanks for your interest in PySeismoSoil!

According to [this place in the code](https://github.com/jsh9/PySeismoSoil/blob/74003861abfd64c1e8727d586b54c801c95b898f/PySeismoSoil/class_simulation.py#L380-L405) or [this page in the documentation](https://pyseismosoil.readthedocs.io/en/stable/api/PySeismoSoil.class_simulation.Nonlinear_Simulation.html) (this URL may get broken in the future), in order to run the HH nonlinear simulation, users would need to curve-fit the in-situ or lab test data to obtain the HH parameters.

This is because the underlying algorithm (forward Euler finite difference) requires parametric representation of the material's stiffness and damping, i.e., we need to use a formula rather than discrete data points.

You can use the equivalent linear method and directly feed your experiment data to it.  But you probably don't want to use equivalent linear because it's not the most accurate numerical method.

(Tagging @domniki for any additional insights.)

**Reply by xzzgameboy** [reply]
Thank you @jsh9 for your insight on this topic. Apprecaite it.
